autometrics/src/otel_push_exporter.rs

```diff
@@ -0,0 +1,93 @@
+use opentelemetry_api::metrics::MetricsError;
+use opentelemetry_otlp::OtlpMetricPipeline;
+use opentelemetry_otlp::{ExportConfig, Protocol, WithExportConfig};
+use opentelemetry_sdk::metrics::MeterProvider;
+use opentelemetry_sdk::runtime;
+use std::ops::Deref;
+use std::time::Duration;
+
+#[repr(transparent)]
+pub struct OtelMeterProvider(MeterProvider);
+
+impl Deref for OtelMeterProvider {
+    type Target = MeterProvider;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl Drop for OtelMeterProvider {
+    fn drop(&mut self) {
+        // this will only error if `.shutdown` gets called multiple times
+        let _ = self.0.shutdown();
+    }
+}
+
+#[cfg(feature = "otel-push-exporter-http")]
+pub fn init_http(url: impl Into<String>) -> Result<OtelMeterProvider, MetricsError> {
+    runtime()
+        .with_exporter(
+            opentelemetry_otlp::new_exporter()
+                .http()
+                .with_export_config(ExportConfig {
+                    endpoint: url.into(),
+                    protocol: Protocol::HttpBinary,
+                    ..Default::default()
+                }),
+        )
+        .with_period(Duration::from_secs(1))
```

Comment:
I think it might be cool to allow users to choose their own `period` and possibly `timeout` int the init call. I suppose that controlling the Period would be nice if you want to limit traffic a bit, not sure about the timeout (I thought "timeout might need to be set according to the `period`", but maybe users don't care